### ISSUE TITLE
[Bug] HTML error: Colgroup displays empty block in table

### ISSUE THREAD

**Issue by tomkahn** [ankidroid/Anki-Android#9488]
###### Reproduction Steps

The latest issue of AnkiDroid has some weird HTML issues that other versions of Anki don't have (and AnkiDroid also didn't have before the latest update from Google Play). You can see what is wrong in the screenshot below. To reproduce the issue you must:
* Have a colgroup-element inside a table
* Set the span count to a number that is bigger than the amount of VISIBLE columns in this table

Example:
```
<table>
	<colgroup>
		<col width=450 span=2> <!-- Cause 1 -->
	</colgroup>
	<tr>
		<td id="Name_1">{{Objekt 1}}</td>
		<td id="Name_2" style="display:none">{{Objekt 2}}</td> <!-- Cause 2 -->
	</tr>
	<tr>
		<td id="Inhalt_1">???</td>
		<td id="Inhalt_2" style="display:none">{{Antwort 2}}</td> <!-- Cause 3 -->
	</tr>
</table>
```

APKG to check for yourself here: https://www.dropbox.com/s/dhpnliees8k5b7x/AnkiDroid_colgroup_span_bug.apkg?dl=0

###### Expected Result

![image](https://user-images.githubusercontent.com/38276038/132065016-7044bd48-bea9-4e13-8964-1f29b68a9bca.png)

###### Actual Result

![image](https://user-images.githubusercontent.com/38276038/132065203-f98ab8e4-a5e1-4a4a-a1bd-e28e9ae9d649.png)

###### Debug info

Please see here: https://www.dropbox.com/s/llkbcgc61jkibp3/ankidroid_info.txt?dl=0

###### Research

- [x] I have searched for similar existing issues here and on the user forum
- [x] I have confirmed the issue is not resolved in the latest alpha release 2.16alpha3

**Comment by david-allison-1** [comment]
Thanks for the great bug report!

Reproduced on AnkiDroid 2.14.6. This could be due to an update with the `Android System WebView` component.

Could you find a parallel release where this works as expected

A sample parallel release would be available under this URL. `AnkiDroid.Parallel.A` is installable alongside AnkiDroid. Note that AnkiDroid 2.15.X locks the collection, so you won't be able to run both at the same time:

https://github.com/ankidroid/Anki-Android/releases/tag/v2.15.6

**Comment by tomkahn** [comment]
You're welcome. Thank you for the fast response.

I just removed the import statement from the CSS and the issue is still there. Also updated [the APKG](https://www.dropbox.com/s/dhpnliees8k5b7x/AnkiDroid_colgroup_span_bug.apkg?dl=0) if you want to try for yourself.

I then continued trying different parallel (A) versions, but so far the same problem occurred in:
* 2.15.6
* 2.15.5
* 2.15.4
* 2.15.3
* 2.15.2
* 2.14.6

This is slightly surprising as neither I nor any of my users noticed it before the latest version. If all of these versions simply use the same Android System WebView, however, it would make sense, though.

**Comment by david-allison-1** [comment]
It'd be easier for you to go back a fair bit (2.10.X for example) and bisect. It's likely a WebView issue, in which case we won't actively be able to do anything in-app to fix this.

I'd advise to check to confirm that your HTML is valid, and that the JS debugging finds no errors in the dev console:

https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#html-javascript-inspection

**Comment by VladThodo** [comment]
I've simply copied the HTML code the Anki  logs when displaying a HTML card (here's the  [HTML](https://gist.github.com/VladThodo/cf8a9585ba577bc5b3c54bb1307bb321)) and put it into a file. I then tried rendering it in both Chromium and Firefox and the result is the same as the one displayed on mobile (see picture below; left - Firefox; right - Chromium). Ironically enough, I tried  it also in AnkiDesktop  and it doesn't display the hidden column. However, I think this is more of a HTML issue since both desktop browsers and the mobile WebView agree on how this should be rendered. Maybe AnkiDesktop is the one displaying it wrong?

![DeepinScreenshot_select-area_20210904201352](https://user-images.githubusercontent.com/17083460/132104388-3908cbe5-da7f-43ce-b4be-1861f169c4f3.png)
